app/uk/gov/hmrc/gform/sharedmodel/form/Form.scala

```diff
@@ -41,7 +41,8 @@ case class Form(
   formData: FormData,
   status: FormStatus,
   visitsIndex: VisitIndex,
-  envelopeExpiryDate: Option[EnvelopeExpiryDate]
+  envelopeExpiryDate: Option[EnvelopeExpiryDate],
+  obligations: Option[ListAllInfo]
```

Comment:
Is there a meaningful difference between None and Some(List.empty) ?  If not, then it should either not be an Option, or should be an Option of a NonEmptyList

Reply:
In fact, this ought not be either a List[], or an Option[NonEmptyList], it should be

obligations: NonEmptyList[TaxPeriodInformation]